### DISCUSSION TITLE
inferred untyped anonymous functions

### DISCUSSION THREAD

**Discussion by shadowninja55** [vlang/v#11887] [Ideas]
when an anonymous function is passed to another function that function already knows the argument and return types. thus, you should be able to omit them from the function definition. this would also allow us to remove some compiler magic and remove some boilerplate. for instance:
`arr.map(fn (it) { it.str() })`
`arr.sort(fn (a, b) { a < b })`
`arr.reduce(fn (a, b) { a * b })`

**Comment by danieldaeschle** [comment]
I'm against this. This add complication without much value

**Comment by shadowninja55** [comment]
> I'm against this. This add complication without much value

in my opinion it can serve as a way to remove the complication of compiler magic (like in sort, map, any, etc) and introduce a non-compiler-magic solution that is still clean.

**Comment by dumblob** [comment]
> in my opinion it can serve as a way to remove the complication of compiler magic (like in sort, map, any, etc) and introduce a non-compiler-magic solution that is still clean.

But do we want this? I for myself don't think compiler magic is any problem.

**Comment by danieldaeschle** [comment]
> in my opinion it can serve as a way to remove the complication of compiler magic (like in sort, map, any, etc) and introduce a non-compiler-magic solution that is still clean.

we have a non compiler magic solution

**Comment by shadowninja55** [comment]
> > in my opinion it can serve as a way to remove the complication of compiler magic (like in sort, map, any, etc) and introduce a non-compiler-magic solution that is still clean.
> 
> we have a non compiler magic solution

sure, but it's completely verbose and in my opinion not something i'd want to be typing.

**Comment by danieldaeschle** [comment]
A language can never statisfy everyones wishes.
